lib/dal/src/component/debug.rs

```diff
@@ -0,0 +1,177 @@
+use petgraph::graph::NodeIndex;
+use petgraph::Direction;
+use serde::{Deserialize, Serialize};
+use std::collections::{HashMap, VecDeque};
+use telemetry::prelude::*;
+use thiserror::Error;
+
+use crate::attribute::value::AttributeValueError;
+
+use crate::attribute::value::debug::AttributeDebugViewError;
+use crate::prop::PropError;
+use crate::workspace_snapshot::edge_weight::EdgeWeightKind;
+use crate::workspace_snapshot::node_weight::NodeWeightError;
+use crate::workspace_snapshot::WorkspaceSnapshotError;
+use crate::{
+    func::execution::FuncExecutionError, AttributeValue, AttributeValueId, Component, ComponentId,
+    DalContext, PropId, SchemaVariantId, SecretError, SecretId,
+};
+use crate::{ComponentError, FuncError, SchemaVariantError};
+
+type ComponentDebugViewResult<T> = Result<T, ComponentDebugViewError>;
+
+/// A generated view for an [`Component`](crate::Component) that contains metadata about each of
+/// the components attributes. Used for constructing a debug view of the component and also for
+/// cloning a component
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ComponentDebugView {
+    pub name: String,
+    pub schema_variant_id: SchemaVariantId,
+    pub attribute_tree: HashMap<AttributeValueId, Vec<AttributeValueId>>,
+    // pub input_sockets: Vec<AttributeDebugView>,
+    // pub output_sockets: Vec<AttributeDebugView>,
```

Comment:
We should delete the commented out stuff unless accompanied by a `NOTE`, `TODO` or `FIXME`.